derive-encode/src/lib.rs

```diff
@@ -22,22 +22,35 @@ pub fn derive_encode_label_set(input: TokenStream) -> TokenStream {
             syn::Fields::Named(syn::FieldsNamed { named, .. }) => named
                 .into_iter()
                 .map(|f| {
-                    let ident = f.ident.unwrap();
-                    let ident_string = KEYWORD_IDENTIFIERS
+                    let flatten = f
+                        .attrs
                         .iter()
-                        .find(|pair| ident == pair.1)
-                        .map(|pair| pair.0.to_string())
-                        .unwrap_or_else(|| ident.to_string());
-
-                    quote! {
-                        let mut label_encoder = encoder.encode_label();
-                        let mut label_key_encoder = label_encoder.encode_label_key()?;
-                        EncodeLabelKey::encode(&#ident_string, &mut label_key_encoder)?;
-
-                        let mut label_value_encoder = label_key_encoder.encode_label_value()?;
-                        EncodeLabelValue::encode(&self.#ident, &mut label_value_encoder)?;
-
-                        label_value_encoder.finish()?;
+                        .find(|a| a.path.is_ident("prometheus"))
+                        .map(|a| a.parse_args::<syn::Ident>().unwrap().to_string() == "flatten")
```

Comment:
Today the derive macro does not support any attributes. With that in mind, what do you think of providing an error message here in case a user provides anything other than `flatten`? E.g. in the case of `flaaaaatten` it would be helpful for the user to get an error message along the lines of: "Provided attribute `flaaaaatten` but only `flatten` is supported.". What do you think?

Reply:
Makes sense. Good idea, thanks!